fdbserver/SimulatedCluster.actor.cpp

```diff
@@ -312,12 +312,29 @@ class TestConfig : public BasicTestConfig {
 			if (attrib == "blobGranulesEnabled") {
 				blobGranulesEnabled = strcmp(value.c_str(), "true") == 0;
 			}
+			if (attrib == "allowDefaultTenant") {
+				allowDefaultTenant = strcmp(value.c_str(), "true") == 0;
+			}
+			if (attrib == "allowDisablingTenants") {
+				allowDisablingTenants = strcmp(value.c_str(), "true") == 0;
+			}
+			if (attrib == "allowCreatingTenants") {
+				allowCreatingTenants = strcmp(value.c_str(), "true") == 0;
+			}
 			if (attrib == "injectSSTargetedRestart") {
 				injectTargetedSSRestart = strcmp(value.c_str(), "true") == 0;
 			}
-
-			if (attrib == "injectSSDelay") {
-				injectSSDelay = strcmp(value.c_str(), "true") == 0;
+			if (attrib == "tenantModes") {
```

Comment:
I originally tried that but the actual enum name and what is provided with `toString` don't match, so writing a `fromString` looked less appealing to me because it would not be clear

Reply:
I think it's going to be helpful to people to be able to see the actual mode being used rather than having to do a manual translation. We can add a new method to help with this, and we should probably use the strings that don't include the experimental suffix (required, optional, disabled).

Reply:
For the record, `storageEngineExcludeTypes` in the test config does the same thing, parsing an array of numbers. I can attempt this change but the actual argument handling and readability may end up slightly clunky. For example, even with your suggestions, `toString` and `fromString` will not translate back-and-forth cleanly without some type of string pre-processing, and parsing an array of strings from .toml will be new as well

Reply:
It's true, and honestly I find that difficult to work with as well. For an example of another field which uses strings for enums, see `extraDatabaseMode`